cirq-core/cirq/devices/named_topologies.py

```diff
@@ -0,0 +1,337 @@
+    of relevant topologies. Relevant topologies may be hardware dependant, but common topologies
+    are linear (1D) and rectangular grid topologies.
+    """
+
+    name: str = NotImplemented
+    """A name that uniquely identifies this topology."""
+
+    n_nodes: int = NotImplemented
+    """The number of nodes in the topology."""
+
+    graph: nx.Graph = NotImplemented
+    """A networkx graph representation of the topology."""
+
+
+_GRIDLIKE_NODE = Union['cirq.GridQubit', Tuple[int, int]]
+
+
+def _node_and_coordinates(
+    nodes: Iterable[_GRIDLIKE_NODE],
+) -> Iterable[Tuple[_GRIDLIKE_NODE, Tuple[int, int]]]:
+    """Yield tuples whose first element is the input node and the second is guaranteed to be a tuple
+    of two integers. The input node can be a tuple of ints or a GridQubit."""
+    for node in nodes:
+        if isinstance(node, GridQubit):
+            yield node, (node.row, node.col)
+        else:
+            x, y = node
+            yield node, (x, y)
+
+
+def draw_gridlike(
+    graph: nx.Graph, ax: plt.Axes = None, tilted: bool = True, **kwargs
+) -> Dict[Any, Tuple[int, int]]:
+    """Draw a Grid-like graph.
+
+    This wraps nx.draw_networkx to produce a matplotlib drawing of the graph.
+
+    Args:
+        graph: A NetworkX graph whose nodes are (row, column) coordinates.
+        ax: Optional matplotlib axis to use for drawing.
+        tilted: If True, directly position as (row, column); otherwise,
+            rotate 45 degrees to accommodate google-style diagonal grids.
+        kwargs: Additional arguments to pass to `nx.draw_networkx`.
+
+    Returns:
+        A positions dictionary mapping nodes to (x, y) coordinates suitable for future calls
+        to NetworkX plotting functionality.
+    """
+    if ax is None:
+        ax = plt.gca()  # coverage: ignore
+
+    if tilted:
+        pos = {node: (y, -x) for node, (x, y) in _node_and_coordinates(graph.nodes)}
+    else:
+        pos = {node: (x + y, y - x) for node, (x, y) in _node_and_coordinates(graph.nodes)}
+
+    nx.draw_networkx(graph, pos=pos, ax=ax, **kwargs)
+    ax.axis('equal')
+    return pos
+
+
+@dataclass(frozen=True)
+class LineTopology(NamedTopology):
+    """A 1D linear topology.
+
+    Node indices are contiguous integers starting from 0 with edges between
+    adjacent integers.
+
+    Args:
+        n_nodes: The number of nodes in a line.
+    """
+
+    n_nodes: int
+
+    def __post_init__(self):
+        if self.n_nodes <= 1:
+            raise ValueError("`n_nodes` must be greater than 1.")
+        object.__setattr__(self, 'name', f'line-{self.n_nodes}')
+        graph = nx.from_edgelist(
+            [(i1, i2) for i1, i2 in zip(range(self.n_nodes), range(1, self.n_nodes))]
+        )
+        object.__setattr__(self, 'graph', graph)
+
+    def draw(self, ax=None, tilted: bool = True, **kwargs) -> Dict[Any, Tuple[int, int]]:
+        """Draw this graph.
```

Comment:
I would add a small bit of detail here, such as `Draw this graph using matplotlib.` so that the docstring is not redundant with the function name.

Reply:
added matplotlib notes here and everywhere